internal/ethapi/simulate.go

```diff
@@ -81,6 +81,21 @@ type simBlockResult struct {
 	Calls       []simCallResult
 }
 
+type nullReceiptGetter struct{}
+
+func (nullReceiptGetter) GetReceipts(context.Context, common.Hash) (types.Receipts, error) {
+	return nil, nil
```

Comment:
From conversation with @sebastianst : let's panic here instead of returning nil.. We actually would return `nil` anyway if it isn't a `DepositTx` type (which is specific to Optimism and wouldn't occur in the upstream geth tests). 

This ensures we aren't assuming that this isn't being used anywhere else.